datastore_entity_derives/src/lib.rs

```diff
@@ -106,72 +139,60 @@ pub fn datastore_managed(input: TokenStream) -> TokenStream {
-                                    datastore_property_name, struct_property_name
-                                );
-                                field_metas.push(FieldMeta {
+                                field_metas.push(build_field_meta(
                                     ident,
-                                    into_property: parse_expr(&into_property_expr_string),
-                                    from_property: parse_expr(&from_property_expr_string),
-                                    entity_getter: generate_entity_getter(
-                                        indexed,
-                                        "i64",
-                                        &struct_property_name,
-                                        &datastore_property_name,
-                                    ),
-                                });
+                                    &datastore_property_name,
+                                    &struct_property_name,
+                                    "integer",
+                                    indexed,
+                                    "i64",
+                                ));
                             }
                             "bool" => {
-                                let into_property_expr_string =
-                                    format!("properties.get_bool(\"{}\")", datastore_property_name);
-                                let from_property_expr_string = format!(
-                                    "properties.set_bool(\"{}\", entity.{})",
-                                    datastore_property_name, struct_property_name
-                                );
-                                field_metas.push(FieldMeta {
+                                field_metas.push(build_field_meta(
                                     ident,
-                                    into_property: parse_expr(&into_property_expr_string),
-                                    from_property: parse_expr(&from_property_expr_string),
-                                    entity_getter: generate_entity_getter(
-                                        indexed,
-                                        "bool",
-                                        &struct_property_name,
-                                        &datastore_property_name,
-                                    ),
-                                });
+                                    &datastore_property_name,
+                                    &struct_property_name,
+                                    "bool",
+                                    indexed,
+                                    "bool",
+                                ));
+                            }
+                            "Vec" => {
+                                match get_generic_argument(p) {
+                                    Some(generic_type) => match generic_type.as_str() {
+                                        "String" => {
```

Comment:
If everything from `match &field.ty` is broken into a `fn` then this could recurse here? Maybe in 1.1

Reply:
Hmm, yes maybe, not sure if it would be super much easier, but yea maybe.
Will look into that when fixing the rest of the types for arrays and Optional - A feature soon in a PR near you